docs/DeveloperGuide.md

```diff
@@ -196,12 +196,54 @@ The following activity diagram summarizes what happens when a user executes a li
+When a user enters a valid Add command in the interface, `AddressBookParser#parseCommand` will be called which processes the inputs, creates an instance of a command parser, and calls the `projectCommandParser#parse`,
+`IssueCommandParser#parse` or `ClientCommandParser#parse` method, depending on which entity is being added. Within this method, the flag `-a` will be detected, calling `ProjectCommandParser#parseAddProjectCommand`, 
+`IssueCommandParser#parseAddIssueCommand`, or `ClientCommandParser#parseAddClientCommand`, depending on which entity is added, which checks for input and prefix-value pair validity with methods in `ParserUtil`.
+Finally, the parsed arguments are passed into and returned in an instance of the Add Command entity and the `AddProjectCommand#execute`, `AddIssueCommand#execute`, or `AddClientCommand#execute` is called depending
+on which entity is added, which retrieves the respective entity list from the system, adds the entity into the list to update it, and have the UI display the updated filtered entity list.   
 
-_{Explain here how the data archiving feature will be implemented}_
+#### Add Project Command
+Compulsory prefixes: n/<valid name>
+Optional prefixes: cid/<valid client id>, r/<valid repository>, d/<valid deadline>
+Example Use: `project -a n/John cid/1 r/JohnDoe/tp d/2022-03-05`
 
+#### Add Issue Command
+Compulsory prefixes: n/<valid name>, desc/<valid description>
+Optional prefixes: pid/<valid project id> d/<valid deadline> p/<valid priority>
+Example Use: `issue pid/1 desc/To create a person class which stores all relevant person data d/2022-12-10 p/0`
+
+#### Add Client Command
+Compulsory prefixes: n/<valid name>, pid/<valid project id>
+Optional prefixes: p/<valid phone>, e/<valid email>
+Example Use: `client -a n/John Doe p/98765432 e/johnd@example.com pid/1`
+
+#### The following sequence diagram shows how the add command operation works for adding a project entity:
+Example: `project -a n/John cid/1 r/JohnDoe/tp d/2022-03-05`
+
+![AddSequenceDiagram](images/AddSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">
+:information_source: **Note:** The lifeline for `AddProjectCommand` 
+should end at destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+#### Design considerations:
+
+**Aspect: Add command access to the model: **
+
+**Alternative 1: ** Refactor `ProjectCommandParser:parseAddProjectCommand`, `IssueCommandParser:parseAddIssueCommand` and `ClientCommandParser:parseAddClientCommand` to have access to the Model.
+ * Pros: Mappings could be performed within the parser which fitted its responsibility.
+ * Cons: May result in extra coupling between Parser class and Model class.
+
+**Alternative 2: (current choice)** Only `AddProjectCommand:execute`, `IssueCommandParser:execute` and `ClientCommandParser:execute` have access to the Model.
```

Comment:
Minor nit - current choice might be better off documented as the first alternative

Reply:
Sure current choice to be the first option